cli/cmd/upgrade.go

```diff
@@ -346,25 +353,26 @@ func (options *upgradeOptions) validateAndBuild(stage string, k kubernetes.Inter
 
 	values.Stage = stage
 
-	// Update Add-Ons Configuration from the linkerd-value cm
-	cmRawValues, _ := k8s.GetAddOnsConfigMap(k, controlPlaneNamespace)
-
-	if cmRawValues != nil {
-		//Cm is present now get the data
-		cmData := cmRawValues["values"]
-		rawValues, err := yaml.Marshal(values)
-		if err != nil {
-			return nil, nil, err
-		}
+	if !options.addOnOverwrite {
+		// Update Add-Ons Configuration from the linkerd-value cm
+		cmRawValues, _ := k8s.GetAddOnsConfigMap(k, controlPlaneNamespace)
+		if cmRawValues != nil {
+			//Cm is present now get the data
+			cmData := cmRawValues["values"]
```

Comment:
Is it safe assuming that this key is present in the map? Is it worth verifying ?

Reply:
It is definitely safe theoretically, but added the verification here. Also, I added the case to fail here if `values` key is not present instead of skipping as it is weird to have `linkerd-config-addons` and not have `values`. @alpeb WDYT?

Reply:
Agreed it's better to fail in those weird cases :+1: